entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -49,26 +51,32 @@ class EntityHandler(
             .map {
                 entityService.createEntity(it)
             }
-            .flatMap {
-                created(URI("/ngsi-ld/v1/entities/${it.id}")).build()
+            .map {
+                ResponseEntity.created(URI("/ngsi-ld/v1/entities/${it.id}")).build<String>()
             }
     }
 
     /**
      * Implements 6.4.3.2 - Query Entities
      */
-    fun getEntities(req: ServerRequest): Mono<ServerResponse> {
-        val type = req.queryParam("type").orElse("")
-        val q = req.queryParams()["q"].orEmpty()
+    @GetMapping(produces = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE])
+    fun getEntities(@RequestHeader httpHeaders: HttpHeaders, @RequestParam params: MultiValueMap<String, String>):
+            Mono<ResponseEntity<*>> {
+        val type = params.getFirst("type") ?: ""

```

Comment:
Maybe all the logic related to request validation could be in a different component so it can be shared if needed ?

Reply:
I'm not sure that this request validation can be used by other handlers methods.
We can move it to the shared lib when needed.

Reply:
Indeed, it is specific (we'll just have more or less the same in temporal queries, but that's another topic)